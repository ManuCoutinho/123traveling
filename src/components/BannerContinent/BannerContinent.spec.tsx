import { screen, render } from '@testing-library/react'
import { BannerContinent } from '.'

const mockProps = {
  continent: 'Lua',
  image: 'lua',
  color: '#fff'
}
describe('<BannerContinent/>', () => {
  it('should render BannerContinent title correctly ', () => {
    render(<BannerContinent {...mockProps} />)
    expect(screen.getByRole('heading', { name: /Lua/ })).toBeInTheDocument()
  })
  it('should render BannerContinent with color prop', () => {
    render(<BannerContinent {...mockProps} />)
    expect(screen.getByRole('heading', { name: /Lua/ })).toHaveStyle('color: #fff')
  })
  it('should render BannerContinent with background image ', () => {
    const { container } = render(<BannerContinent {...mockProps} />)
    expect(container.firstChild).toHaveStyle('background-image: url("/assets/continents/lua.jpg")')
  })
  it('should BannerContinent match to snapshot ', () => {
    const { container } = render(<BannerContinent {...mockProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
