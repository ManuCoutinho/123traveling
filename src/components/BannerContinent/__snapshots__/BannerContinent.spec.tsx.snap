// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BannerContinent/> should BannerContinent match to snapshot  1`] = `
<div
  class="css-rdxw7m"
>
  <h2
    class="chakra-heading css-1in2q97"
  >
    Lua
  </h2>
</div>
`;
